hack/local-development/start-gardenlet

```diff
@@ -100,6 +100,21 @@ if [ -z "${SEED_KUBECONFIG:-}" ]; then
     base64 -d > "$SEED_KUBECONFIG"
 fi
 
+# apply RBAC resources in seed cluster
+GARDENLET_CHARTS_DIR=charts/gardener/gardenlet
+GARDENLET_CHARTS_RUNTIME_DIR="$GARDENLET_CHARTS_DIR/charts/runtime"
+helm template gardenlet "$GARDENLET_CHARTS_RUNTIME_DIR" -s templates/clusterrole-gardenlet.yaml -f "$GARDENLET_CHARTS_DIR/values.yaml" | kubectl --kubeconfig="$SEED_KUBECONFIG" apply -f -
+helm template gardenlet "$GARDENLET_CHARTS_RUNTIME_DIR" -s templates/clusterrolebinding-gardenlet.yaml -f "$GARDENLET_CHARTS_DIR/values.yaml" | kubectl --kubeconfig="$SEED_KUBECONFIG" apply -f -
+helm template gardenlet "$GARDENLET_CHARTS_RUNTIME_DIR" -s templates/serviceaccount.yaml -f "$GARDENLET_CHARTS_DIR/values.yaml" | kubectl --kubeconfig="$SEED_KUBECONFIG" apply -f -
```

Comment:
I see 3 different problems here:

1. For RBAC resources it is better to use `kubectl auth reconcile` instead of `kubectl apply`.
Also, flags like `--remove-extra-permissions` and `--remove-extra-subjects` should be considered.

2. Using `-s` seems fine, however if a new file with RBAC resource is introduced, it must be set also here and this can be easily missed. Therefore, I think it is better to render the entire chart and then select only the relevant resources, something like 
```bash
helm template gardenlet "$GARDENLET_CHARTS_RUNTIME_DIR"  | yaml2json  | jq 'select(.apiVersion=="rbac.authorization.k8s.io/v1")'
```


3. Using `$GARDENLET_CHARTS_DIR/values.yaml` looks incorrect to me as seed specific configurations are actually in the seed specific gardenlet config file `$configFile` where different feature gates are enabled/disabled.

Reply:
Regarding 3, there are very few values that are currently used in the actual templates that are rendered, and none of the feature gates or other stuff that might be configured in the `$configFile`. Having in mind what's inside these resources, I think the probability of this changing in the future is pretty low. Therefore, I would keep this as it is for the sake of simplicity.
The other 2 suggestions should definitely be considered, thanks!

Reply:
> Regarding 3, there are very few values that are currently used in the actual templates that are rendered, and none of the feature gates or other stuff that might be configured in the `$configFile`. Having in mind what's inside these resources, I think the probability of this changing in the future is pretty low. Therefore, I would keep this as it is for the sake of simplicity.

Why are you saying this? Just take a look at https://github.com/gardener/gardener/blob/919e52dc75b84dcc95717cd80f24b5fe0194262f/charts/gardener/gardenlet/charts/runtime/templates/clusterrole-managed-istio.yaml#L2-L3 which now rings another bell - not all relevant RBAC resources are applied.


I am now concerned how this PR was tested before to be opened without `gardener.cloud:system:gardenlet:managed-istio` clusterrole and `gardener.cloud:system:gardenlet` role being applied?

Reply:
Sorry, my bad, somehow I missed this file.

> I am now concerned how this PR was tested before ...

I didn't test it myself, relied on the author, obviously I will have to do it.

Reply:
The proposed changes are applied. In regard of 3 - new temp file is generated which is result of concatenating the `$configFile` and `values.yaml`. Now it seems gardenlet to work correct and reconciliation of a shoot is successful.